pkg/blob/blob.go

```diff
@@ -693,6 +694,37 @@ func getStorageAccount(secrets map[string]string) (string, string, error) {
 	return accountName, accountKey, nil
 }
 
+// get account sas token if existed from secrets map
+func getAccountSasToken(secrets map[string]string, accountName string) (string, error) {
+	if secrets == nil {
+		return "", fmt.Errorf("unexpected: getAccountSasToken secrets is nil")
+	}
+
+	var account, accountSasToken string
+	for k, v := range secrets {
+		v = strings.TrimSpace(v)
+		switch strings.ToLower(k) {
+		case accountNameField:
+			account = v
+		case defaultSecretAccountName: // for compatibility with built-in azurefile plugin
+			account = v
+		case accountSasTokenField:
+			accountSasToken = v
+		}
+	}
+
+	if account != accountName {
+		return "", fmt.Errorf("storage account(%s) from secret is not %s", account, accountName)
+	}
+
+	if accountSasToken == "" {
+		return accountSasToken, fmt.Errorf("could not find %s field secrets(%v)", accountSasTokenField, secrets)
+	}
+
+	klog.V(4).Infof("got sas token(%s) of storage account(%s) from secret", accountSasToken, accountName)
```

Comment:
don't print the token value, that's leaking credentials

Reply:
done